src/app/pipes/date-format.pipe.ts

```diff
@@ -7,18 +7,22 @@ import * as moment from 'moment';
 })
 export class DateFormatPipe implements PipeTransform {
 
-  defaultDateFormat: string;
-
   constructor(private settingsService: SettingsService) {
-    this.defaultDateFormat = this.settingsService.dateFormat;
-    this.defaultDateFormat = this.defaultDateFormat.replace('dd', 'DD');
   }
 
-  transform(value: Date | moment.Moment, dateFormat: string): any {
+  transform(value: any, dateFormat: string): any {
+    const defaultDateFormat = this.settingsService.dateFormat.replace('dd', 'DD');
+    if (typeof value === 'undefined') {
+      return "";
+    }
+    let dateVal = moment(value);
+    if (value instanceof Array) {
```

Comment:
Wouldnt it be better to:
```
let dateVal;
if(value instanceof Array) {
    dateVal = moment(value.join("-"), "YYYY-MM-DD");
}  else {
    dateVal = moment(value);
}
```

This way if the `value` was an Array, it will not even try to convert to Moment.
@josehernandezfintecheandomx  what do you think?

Reply:
Done!